### PR TITLE
[INLONG-6054][Manager] Kafka sink supports all migration

### PR DESCRIPTION
**Pull request by Yizhou-Yang**
Fixes: #6054

compiled, need further testing to match whole database kafka

### CODE REVIEW

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sink/kafka/KafkaSink.java [hunk 1]
```diff
@@ -61,6 +61,18 @@ public class KafkaSink extends StreamSink {
     @ApiModelProperty("Primary key is required when serializationType is json, avro")
     private String primaryKey;
 
+    @ApiModelProperty("the database-table mapping rule")
+    private String topicPattern;
+
+    @ApiModelProperty(value = "Automatically create kafka topic or not, a note for users.")
+    private String autoCreateTopics;
+
+    @ApiModelProperty(value = "the partition strategy for kafka")
```
Comment (healchow) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sink/kafka/KafkaSink.java:70]:
```suggestion
    @ApiModelProperty(value = "Partition strategy")
```

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sink/kafka/KafkaSink.java [hunk 2]
```diff
@@ -61,6 +61,18 @@ public class KafkaSink extends StreamSink {
...
+    private String partitionStrategy;
+
+    @ApiModelProperty("data multiple format,only applicable when outer format is raw")
```
Comment (healchow) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sink/kafka/KafkaSink.java:73]: What does the `data multiple format` mean?

**Review by healchow — CHANGES_REQUESTED** [review]

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sink/kafka/KafkaSink.java (hunk 2 above)
Comment (Yizhou-Yang) [reply inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sink/kafka/KafkaSink.java:73]: for debezium and canal, the outer format is "raw" , and then sinkmultipleformat is "debezium"/"canal"

Comment (Yizhou-Yang) [reply inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sink/kafka/KafkaSink.java:73]: used by sort connector to parse debezium/canal json

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sink/kafka/KafkaSink.java [hunk 3]
```diff
@@ -61,6 +61,18 @@ public class KafkaSink extends StreamSink {
     @ApiModelProperty("Primary key is required when serializationType is json, avro")
     private String primaryKey;
 
+    @ApiModelProperty("the database-table mapping rule")
```
[the remark this answers is not in the document]
Comment (Yizhou-Yang) [reply inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sink/kafka/KafkaSink.java]: changed

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sink/kafka/KafkaSink.java [hunk 4]
```diff
@@ -61,6 +61,18 @@ public class KafkaSink extends StreamSink {
...
+    private String topicPattern;
+
+    @ApiModelProperty(value = "Automatically create kafka topic or not, a note for users.")
```
[the remark this answers is not in the document]
Comment (Yizhou-Yang) [reply inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sink/kafka/KafkaSink.java]: changed

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sink/kafka/KafkaSinkDTO.java [hunk 5]
```diff
@@ -54,9 +54,21 @@ public class KafkaSinkDTO {
             notes = "including earliest, latest (the default), none")
     private String autoOffsetReset;
 
+    @ApiModelProperty(value = "Automatically create kafka topic or not. values: true, false.")
+    private String autoCreateTopics;
```
Comment (yunqingmoswu) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sink/kafka/KafkaSinkDTO.java:58]: It is not necessary to put this parameter in the manager.

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sink/kafka/KafkaSinkRequest.java [hunk 6]
```diff
@@ -56,4 +56,16 @@ public class KafkaSinkRequest extends SinkRequest {
     @ApiModelProperty("Primary key is required when serializationType is json, avro")
     private String primaryKey;
 
+    @ApiModelProperty("the topic mapping rule")
+    private String topicPattern;
+
+    @ApiModelProperty(value = "Automatically create kafka topic or not. values: true, false.")
+    private String autoCreateTopics;
```
Comment (yunqingmoswu) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sink/kafka/KafkaSinkRequest.java:63]: It is not necessary to put this parameter in the manager.

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/util/LoadNodeUtils.java [hunk 7]
```diff
@@ -162,29 +166,58 @@ public static KafkaLoadNode createLoadNode(KafkaSink kafkaSink, List<FieldInfo>
                 format = new JsonFormat();
                 break;
             case CANAL:
-                format = new CanalJsonFormat();
+                format = new RawFormat();
```
Comment (yunqingmoswu) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/util/LoadNodeUtils.java:169]: This is a error logic.

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/util/LoadNodeUtils.java [hunk 8]
```diff
@@ -162,29 +166,58 @@ public static KafkaLoadNode createLoadNode(KafkaSink kafkaSink, List<FieldInfo>
...
+                innerFormat = new CanalJsonFormat();
                 break;
             case DEBEZIUM_JSON:
-                format = new DebeziumJsonFormat();
+                format = new RawFormat();
```
Comment (yunqingmoswu) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/util/LoadNodeUtils.java:173]: This is a error logic.

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/util/LoadNodeUtils.java [hunk 9]
```diff
@@ -162,29 +166,58 @@ public static KafkaLoadNode createLoadNode(KafkaSink kafkaSink, List<FieldInfo>
...
+                innerFormat = new DebeziumJsonFormat();
                 break;
             default:
                 throw new IllegalArgumentException(String.format("Unsupported dataType=%s for Kafka", dataType));
         }
 
+        String sinkPartitioner = null;
+        if (dataType == DataTypeEnum.CANAL || dataType == DataTypeEnum.DEBEZIUM_JSON) {
+            sinkPartitioner = kafkaSink.getPartitionStrategy() == null ? null : RAW_HASH;
```
Comment (yunqingmoswu) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/util/LoadNodeUtils.java:182]: This is a error logic.

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/util/LoadNodeUtils.java [hunk 10]
```diff
@@ -162,29 +166,58 @@ public static KafkaLoadNode createLoadNode(KafkaSink kafkaSink, List<FieldInfo>
...
+
         return new KafkaLoadNode(
                 kafkaSink.getSinkName(),
                 kafkaSink.getSinkName(),
                 fieldInfos,
                 fieldRelations,
                 Lists.newArrayList(),
                 null,
-                kafkaSink.getTopicName(),
+                "mock_topic",
```
Comment (yunqingmoswu) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/util/LoadNodeUtils.java:209]: Maybe it is not necessary to add a default for the topic?

Comment (Yizhou-Yang) [reply inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/util/LoadNodeUtils.java:209]: topicname is a nonnull parameter, so this must not be null

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sink/kafka/KafkaSink.java [hunk 11]
```diff
@@ -61,6 +61,15 @@ public class KafkaSink extends StreamSink {
     @ApiModelProperty("Primary key is required when serializationType is json, avro")
     private String primaryKey;
 
+    @ApiModelProperty("the topic mapping rule")
```
Comment (healchow) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sink/kafka/KafkaSink.java:64]: `the topic mapping rule` -> `Topic mapping rule`.

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sink/kafka/KafkaSink.java (hunk 2 above)
Comment (healchow) [reply inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sink/kafka/KafkaSink.java:70]: What is the outer format? I didn't see it in KafkaSink.

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/util/LoadNodeUtils.java [hunk 12]
```diff
@@ -167,24 +170,64 @@ public static KafkaLoadNode createLoadNode(KafkaSink kafkaSink, List<FieldInfo>
...
+        Format innerFormat = null;
+        String sinkPartitioner = null;
+        if (dataType == DataTypeEnum.RAW) {
+            switch (innerDataType) {
+                case DEBEZIUM_JSON:
+                    innerFormat = new DebeziumJsonFormat();
+                    break;
+                case CANAL:
+                    innerFormat = new CanalJsonFormat();
+                    break;
```
Comment (healchow) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/util/LoadNodeUtils.java:190]: Please add a default statement for `switch`.

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sink/kafka/KafkaSinkRequest.java [hunk 13]
```diff
@@ -56,4 +56,13 @@ public class KafkaSinkRequest extends SinkRequest {
     @ApiModelProperty("Primary key is required when serializationType is json, avro")
     private String primaryKey;
 
+    @ApiModelProperty("Topic mapping rule")
+    private String topicPattern;
+
+    @ApiModelProperty("Partition strategy")
+    private String partitionStrategy;
+
+    @ApiModelProperty("data multiple format,only applicable when outer format is raw")
```
Comment (healchow) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sink/kafka/KafkaSinkRequest.java:65]:
```suggestion
    @ApiModelProperty("Multiple formats for data content, only needed when the format is 'raw'")
```

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/util/LoadNodeUtils.java (hunk 10 above)
Comment (yunqingmoswu) [reply inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/util/LoadNodeUtils.java]: You can do it in upstream services for now.

**Pull request closed without merging**